File: app/src/main/java/com/jcr/sharedtasks/ui/createproject/CreateProjectViewModel.java
package com.jcr.sharedtasks.ui.createproject;

import android.arch.lifecycle.ViewModel;

import com.jcr.sharedtasks.model.Project;
import com.jcr.sharedtasks.repository.ProjectsRepository;

import java.util.UUID;

import javax.inject.Inject;

public class CreateProjectViewModel extends ViewModel {

    private final ProjectsRepository repository;

    @Inject
    public CreateProjectViewModel(ProjectsRepository projectsRepository) {
        this.repository = projectsRepository;
    }

    public String createProject(String projectName) {
        if (projectName.isEmpty()) return null;
        String projectUUID = UUID.randomUUID().toString();
        Project project = new Project(projectUUID, projectName);
        repository.createProject(project);
        return projectUUID;
    }
}

File: app/src/main/java/com/jcr/sharedtasks/ui/createproject/CreateProjectViewModel.kt
package com.jcr.sharedtasks.ui.createproject

import androidx.lifecycle.ViewModel

import com.jcr.sharedtasks.model.Project
import com.jcr.sharedtasks.repository.ProjectsRepository

import java.util.UUID

import javax.inject.Inject

class CreateProjectViewModel @Inject
constructor(private val repository: ProjectsRepository) : ViewModel() {

    fun createProject(projectName: String): String? {
        if (projectName.isEmpty()) return null
        val projectUUID = UUID.randomUUID().toString()
        val project = Project(projectUUID, projectName)
        repository.createProject(project)
        return projectUUID
    }
}
